src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -2109,6 +2109,43 @@ public FixableDiagnostic ProviderDeclarationViaImportKeywordIsDeprecated(Provide
                 TextSpan,
                 "BCP386",
                 $@"The decorator ""{decoratorName}"" may not be used on statements whose declared type is a reference to a resource-derived type.");
+
+            public ErrorDiagnostic NegatedTypeIndexSought() => new(
+                TextSpan,
+                "BCP387",
+                "Indexing into a type requires a natural number or 0. Unary operators are not permitted.");
```

Comment:
Could we make this message more user friendly?
* "natural number or 0" -> "an integer greater than or equal to 0"
* "Unary operators are not permitted": I suspect many users won't know what a "unary operator" is. Does this need to be said if we've already mentioned >= 0?